File: backend/schemas/employee.py
from pydantic import BaseModel, EmailStr
from typing import Optional, List
from datetime import datetime, date

class EmployeeBase(BaseModel):
    email: EmailStr
    first_name: str
    last_name: str
    employee_code: Optional[str] = None
    phone: Optional[str] = None
    date_of_birth: Optional[date] = None
    date_of_joining: Optional[date] = None
    employment_status: str = "active"
    designation_id: Optional[int] = None
    employment_type: str = "full-time"
    role: str = "employee"
    is_active: bool = True
    gender: Optional[str] = None
    blood_group: Optional[str] = None
    marital_status: Optional[str] = None
    personal_email: Optional[EmailStr] = None
    
    # Emergency Contact
    emergency_contact_name: Optional[str] = None
    emergency_contact_relation: Optional[str] = None
    emergency_contact_phone: Optional[str] = None
    
    department_id: Optional[int] = None
    manager_id: Optional[int] = None
    
    # Banking Information
    bank_account_holder_name: Optional[str] = None
    bank_name: Optional[str] = None
    branch_name: Optional[str] = None
    account_number: Optional[str] = None
    ifsc_code: Optional[str] = None
    account_type: Optional[str] = None
    
    # Statutory Information
    pan_number: Optional[str] = None
    aadhaar_number: Optional[str] = None
    uan_number: Optional[str] = None
    esic_number: Optional[str] = None
    
    profile_photo: Optional[str] = None

class EmployeeCreate(EmployeeBase):
    password: str

class BankingInfoUpdate(BaseModel):
    bank_account_holder_name: str
    bank_name: str
    branch_name: str
    account_number: str
    ifsc_code: str
    account_type: str

class EmployeeUpdate(BaseModel):
    first_name: Optional[str] = None
    last_name: Optional[str] = None
    phone: Optional[str] = None
    date_of_birth: Optional[date] = None
    gender: Optional[str] = None
    blood_group: Optional[str] = None
    marital_status: Optional[str] = None
    personal_email: Optional[EmailStr] = None
    date_of_joining: Optional[date] = None
    
    # Emergency Contact
    emergency_contact_name: Optional[str] = None
    emergency_contact_relation: Optional[str] = None
    emergency_contact_phone: Optional[str] = None
    
    # Admin only fields (should be handled by RBAC in service/API)
    designation_id: Optional[int] = None
    department_id: Optional[int] = None
    manager_id: Optional[int] = None
    employment_type: Optional[str] = None
    employment_status: Optional[str] = None
    employee_code: Optional[str] = None
    is_active: Optional[bool] = None
    
    # Banking
    bank_account_holder_name: Optional[str] = None
    bank_name: Optional[str] = None
    branch_name: Optional[str] = None
    account_number: Optional[str] = None
    ifsc_code: Optional[str] = None
    account_type: Optional[str] = None
    
    # Statutory
    pan_number: Optional[str] = None
    aadhaar_number: Optional[str] = None
    uan_number: Optional[str] = None
    esic_number: Optional[str] = None

class EmployeeResponse(EmployeeBase):
    id: int
    created_at: datetime
    updated_at: Optional[datetime] = None
    full_name: str
    archived_at: Optional[datetime] = None

    class Config:
        from_attributes = True

class EmployeeListResponse(BaseModel):
    items: List[EmployeeResponse]
    total: int
    page: int
    size: int

File: frontend/types/employee.ts

export interface Employee {
    id: number;
    employee_code?: string;
    first_name: string;
    last_name: string;
    email: string;
    phone?: string;
    date_of_birth?: string;
    date_of_joining?: string;
    employment_status: string;
    department_id?: number;
    designation_id?: number;
    manager_id?: number;
    employment_type: string;
    role: string;
    is_active: boolean;
    gender?: string;
    blood_group?: string;
    marital_status?: string;
    personal_email?: string;

    // Emergency Contact
    emergency_contact_name?: string;
    emergency_contact_relation?: string;
    emergency_contact_phone?: string;

    // Banking Information
    bank_account_holder_name?: string;
    bank_name?: string;
    branch_name?: string;
    account_number?: string;
    ifsc_code?: string;
    account_type?: string;

    // Statutory Information
    pan_number?: string;
    aadhaar_number?: string;
    uan_number?: string;
    esic_number?: string;

    profile_photo?: string;
    archived_at?: string;
    created_at: string;
    updated_at?: string;
    full_name: string;
}

export interface EmployeeListResponse {
    items: Employee[];
    total: number;
    page: number;
    size: number;
}
